test/example/strings/demo_padl.f90

```diff
@@ -0,0 +1,15 @@
+program demo_padl
+    use stdlib_string_type, only: string_type, assignment(=)
```

Comment:
This demo is falling in the CI, because is missing a `write(formatted)`.

```suggestion
    use stdlib_string_type, only: string_type, assignment(=), write(formatted)
```